### ISSUE TITLE
Powershell randomly crash [IPC][linux]

### ISSUE THREAD

**Issue by ljubingemy** [PowerShell/PowerShell#24184]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

We use PowerShell to perform Centreon checks for our monitoring solution. However, we are currently experiencing issues where some PowerShell checks randomly stop working

### Expected behavior

```console
(7.4.4:C:80) [NamedPipeIPC_ServerListenerStarted:NamedPipe.Open.Informational] PowerShell has started an IPC listening thread on process: 3789076 in AppDomain: None.
```

### Actual behavior

```console
(7.4.4:C:80) [NamedPipeIPC_ServerListenerError:NamedPipe.Exception.Error] An error has occurred in PowerShell  IPC listening thread on process: 3788629 in AppDomain: None.  Error Message: Operation canceled.
```

### Error details

```console
powershell randomly crash during IPC listening thread
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.4
PSEdition                      Core
GitCommitId                    7.4.4
OS                             AlmaLinux 8.9 (Midnight Oncilla)
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

# Powershell randomly crash on Almalinux

Hello,
We are opening a ticket  because we experimented some problem with Powershell

## Os info:
•	Virtualization: vmware
•	Operating System: AlmaLinux 8.9 (Midnight Oncilla)
•	CPE OS Name: cpe:/o:almalinux:almalinux:8::baseos
•	Kernel: Linux 4.18.0-513.24.1.el8_9.x86_64
•	Architecture: x86-64
•	Powershell info :

![image](https://github.com/user-attachments/assets/18be75a5-24f3-4dd9-9575-b1ad048de589)

## Problem description :
“We use PowerShell to perform Centreon checks for our monitoring solution. However, we are currently experiencing issues where some PowerShell checks randomly stop working

## Log information:
![image](https://github.com/user-attachments/assets/fa2fce80-682b-452c-8972-d6b02c709e9c)

**Comment by GlacierFox** [comment]
Did you find a solution for this @ljubingemy ? I'm getting the exact same error trying to call a PowerShell script from an ASP.Net API application. Runs fine if run it manually which is strange.

**Comment by SeeminglyScience** [comment]
There's not enough information in this issue. If you've determined this is due to a bug in PowerShell itself, please provide steps we can use to reproduce the issue outside of your environment.

If that has not been determined, you would need to reach out to the support for the software you're using or you could try asking the community for assistance in the [PowerShell community discord server](https://aka.ms/psdiscord).

----

@GlacierFox the error message they specified is a very generic one that can be thrown from a lot of different APIs. The answer will likely be something in your code that needs to change to adapt to being ran from ASP.NET.

**Comment by GlacierFox** [comment]
@SeeminglyScience You're right, I figured it out on my end. The error still shows in my system logs but I realised it wasn't the cause of my issues so it's not blocking anything. 

Just to state what was up on my end, quite simply, I was having trouble running a script from an ASP.NET API app on RHEL which contained a session to a different machine over ssh.
I hadn't realised the web API was running the script as root and had no access to my local user authentication keys.
